### PR TITLE
Cryptocurrency

### PR DESCRIPTION
**Pull request by SimonGrchevski**
- Add the React Components
- Add the Redux logic 
- Add the styles
- Use MSW to intercept the API calls, and Add All the tests
- Write the README.md

### CODE REVIEW

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please try to add proper tests in your project. This is one of the basic requirements for the project. 
- [ ] It would begreat if you can add filter stateless. Please check the requirements once again. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by SimonGrchevski** [comment]
> # Changes Requested,
> Hi there,
> 
> Good job so far!
> There are some issues that you still need to work on to go to the next project but you are almost there!
> 
> ## Required Changes 
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> You can also consider:
> 
> * [ ]  Please try to add proper tests in your project. This is one of the basic requirements for the project.
> * [ ]  It would begreat if you can add filter stateless. Please check the requirements once again.
> 
> Cheers and Happy coding!
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SimonGrchevski** [comment]
Note:
I see no comments therefore please consider to kindly review.

**Comment by arslanbisharat** [comment]
> Note:
> I see no comments therefore please consider to kindly review.

Please read my comment once again.

**Comment by SimonGrchevski** [comment]
I did it. There is no comment.

**Comment by arslanbisharat** [comment]
> I did it. There is no comment.

Please try to add proper tests in your project. This is one of the basic requirements for the project. It would begreat if you can add filter stateless. Please check the requirements once again.

**Comment by SimonGrchevski** [comment]
They are all labeled as Optional.

" You can also consider:

 Please try to add proper tests in your project. This is one of the basic requirements for the project.
 It would begreat if you can add filter stateless. Please check the requirements once again. "

**Comment by arslanbisharat** [comment]
These are not optional. I cannot use comments for these requirements so
these are in this section. Hope this helps.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hi there, 

Awesome job on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet. Let's work on those.


 ## Required Changes ♻️
_Kindly check the comments under the review._


Happy coding, Cheers!🚀🚀

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it's valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/MainPage.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  return (
+    <section>
+      <Header currentView="Top ranked crypto values" />
+      {getHighest(state)}
+      <Breakdown text="Stats by rank" />
+      <ul className="crypto-ul">
+        {crypto}
+      </ul>
+    </section>
+  );
```
Comment (mikenath223) [inline src/Components/MainPage.js]: - [x] As requested in the previous review please add the filter component that should enable the user to filter the data retrieved by different values.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "react/jsx-filename-extension": ["warn", { "extensions": [".js", ".jsx"] }],
+    "react/react-in-jsx-scope": "off",
+    "import/no-unresolved": "off",
+    "no-shadow": "off",
+    "import/no-extraneous-dependencies": [
+      "error",
+      {
+        "devDependencies": true
+      }
+    ]
```
Comment (mikenath223) [inline .eslintrc.json]: - [x] Kindly use the same eslint rules listed [here](https://github.com/microverseinc/linters-config/tree/master/react-redux).

File: src/Components/CurrencyDetail.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        <div>
+          <p className="crypto-name">
+            {key}
+            {' '}
+          </p>
+        </div>
+        <div>
+          {value}
+          {' '}
+        </div>
```
Comment (mikenath223) [inline src/Components/CurrencyDetail.js:18]: - [x] OPTIONAL - Consider adding margin/padding to the details listed.

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @SimonGrchevski,

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :male_detective: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nidalaa** [comment]
Hello @SimonGrchevski ,

_This message is a result of the second opinion request_

Due to the bug, your request was lost and restored only today. We apologize for the inconvenience.

As I can see - you and Code Reviewers managed to clarify requirements. Good job here!

**Pull request merged**